ConfigurationSystem/private/ConfigurationData.py

```diff
@@ -332,13 +331,12 @@ def __backupCurrentConfiguration( self, backupName ):
     backupFile = os.path.join( backupPath, configurationFilename.replace( ".cfg", ".%s.zip" % backupName ) )
     if os.path.isfile( configurationFile ):
       gLogger.info( "Making a backup of configuration in %s" % backupFile )
-      try:
-        zf = zipfile.ZipFile( backupFile, "w", zipfile.ZIP_DEFLATED )
-        zf.write( configurationFile, "%s.backup.%s" % ( os.path.split( configurationFile )[1], backupName ) )
-        zf.close()
-      except Exception:
-        gLogger.exception()
-        gLogger.error( "Cannot backup configuration data file", "file %s" % backupFile )
+      with zipfile.ZipFile( backupFile, "w", zipfile.ZIP_DEFLATED ) as zf:
```

Comment:
The `try` has to come before the `with` statement, the constructor can throw an exception.

(I can't find this in the [documentation](https://docs.python.org/2/library/zipfile.html) but a quick test shows this:
```
with zipfile.ZipFile( '/myzip.zip', "w", zipfile.ZIP_DEFLATED ) as zf:
  try:
     zf.write('eggs.txt')
   except Exception:
     print 'exception caught'
```
 
> Traceback (most recent call last):
>  File "<stdin>", line 1, in <module>
>  File "/afs/cern.ch/user/j/jebbing/dirac_home/Linux_x86_64_glibc-2.12/lib/python2.7/zipfile.py", line 756, in >__init__
>    self.fp = open(file, modeDict[mode])
>IOError: [Errno 13] Permission denied: '/myzip.zip' )